tests/test_create.py

```diff
@@ -1282,12 +879,13 @@ def test_install_only_deps_flag(self):
 
     def test_install_update_deps_only_deps_flags(self):
         with make_temp_env("flask=2.0.1", "jinja2=3.0.1") as prefix:
-            assert package_is_installed(prefix, "python=3.9")
+            python = join(prefix, PYTHON_BINARY)
+            result_before = subprocess_call_with_clean_env([python, "--version"])
             assert package_is_installed(prefix, "flask=2.0.1")
             assert package_is_installed(prefix, "jinja2=3.0.1")
-
             run_command(Commands.INSTALL, prefix, "flask", "python=3.9", "--update-deps", "--only-deps")
```

Comment:
This test is not guaranteed to pass, sadly.

The solver might choose a different version before (so not necessarily 3.9), and then we force a 3.9 install. I think it could work with this change:

```diff
- run_command(Commands.INSTALL, prefix, "flask", "python=3.9", "--update-deps", "--only-deps")
+ run_command(Commands.INSTALL, prefix, "flask", "python", "--update-deps", "--only-deps")
```

cc @jezdez

Reply:
You're right, I missed this python=3.9, good catch! I've opened https://github.com/conda/conda/pull/11273.